client/src/pages/lessons/index.tsx

```diff
@@ -0,0 +1,304 @@
+import { withPrefix } from "gatsby";
+import React from "react";
+import {
+  MuiThemeProvider,
+  createMuiTheme,
+  makeStyles,
+} from "@material-ui/core/styles";
+import {
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TablePagination,
+  TableSortLabel,
+  TableRow,
+  Button,
+  IconButton,
+  Toolbar,
+  AppBar,
+} from "@material-ui/core";
+import LaunchIcon from "@material-ui/icons/Launch";
+import AddIcon from "@material-ui/icons/Add";
+import KeyboardArrowLeftIcon from "@material-ui/icons/KeyboardArrowLeft";
+import KeyboardArrowRightIcon from "@material-ui/icons/KeyboardArrowRight";
+import { Link, navigate } from "@reach/router";
+
+import { LessonEdge, LessonsData } from "types";
+import { fetchLessons, createLesson } from "api";
+import NavBar from "components/nav-bar";
+import "styles/layout.css";
+
+const theme = createMuiTheme({
+  palette: {
+    primary: {
+      main: "#1b6a9c",
+    },
+  },
+});
+
+const columns: ColumnDef[] = [
+  { id: "name", label: "Lesson", minWidth: 170, align: "center" },
+  { id: "updatedAt", label: "Date", minWidth: 170, align: "center" },
+];
+
+interface ColumnDef {
+  id: string;
+  name?: string;
+  label: string;
+  minWidth: number;
+  align?: "right" | "left" | "center";
+  format?: (v: number) => string;
+}
+
+const useStyles = makeStyles({
+  root: {
+    width: "100%",
+  },
+  container: {
+    maxHeight: 440,
+  },
+  button: {
+    margin: theme.spacing(1),
+  },
+  appBar: {
+    height: "10%",
+    top: "auto",
+    bottom: 0,
+  },
+});
+
+export const LessonsTable = (props: { location: any }) => {
+  const classes = useStyles();
+  const initialLessons = {
+    edges: [
+      {
+        cursor: "",
+        node: {
+          id: "",
+          lessonId: "",
+          name: "",
+          intro: "",
+          question: "",
+          conclusion: [""],
+          expectations: [
+            {
+              expectation: "",
+              hints: [{ text: "" }],
+            },
+          ],
+          updatedAt: 0,
+          createdAt: 0,
+        },
+      },
+    ],
+    pageInfo: {
+      hasNextPage: false,
+      endCursor: "",
+    },
+  };
+
+  const [lessons, setLessons] = React.useState<LessonsData>(initialLessons);
+  const [prevPages, setPrevPages] = React.useState<string[]>([""]);
+  const [page, setPage] = React.useState(0);
+  const [sortBy, setSortBy] = React.useState("updatedAt");
+  const [sortDesc, setSortDesc] = React.useState(true);
+  const rowsPerPage = 10;
+
+  React.useEffect(() => {
+    let mounted = true;
+    fetchLessons(rowsPerPage, prevPages[page], sortBy, sortDesc)
+      .then((lessons) => {
+        console.log(`fetchLessons got`, lessons);
+        if (mounted) {
+          if (lessons !== undefined) {
```

Comment:
prefer to just use js `falsy`ness for tests and also don't see a need for two separate `if` blocks here so all in all could be

```typescript
if (mounted && lessons) {
```